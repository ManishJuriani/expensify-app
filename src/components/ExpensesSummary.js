import React from 'react';
import numeral from 'numeral';
import { connect } from 'react-redux';
import selectExpensesTotal from "../selectors/expenses-total";
import selectExpenses from '../selectors/expenses';

export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount===1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(props.expensesTotal).format('$0,0.00');
    return (
    <div>
        {   
            <h3>Viewing {props.expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h3>
        }
    </div>
)}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses,state.filters);
    const expenseCount = visibleExpenses.length;
    const expensesTotal = selectExpensesTotal(visibleExpenses);
    return {
        expenseCount,
        expensesTotal
    }
}

export default connect(mapStateToProps)(ExpensesSummary);