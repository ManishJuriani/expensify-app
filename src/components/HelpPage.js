import React from 'react';

const HelpPage = () => (
    <div>
        How may i help you??
    </div>
)

export default HelpPage;