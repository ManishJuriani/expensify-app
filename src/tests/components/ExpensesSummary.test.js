import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should render ExpensesSummary with one expense correctly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={500} />);
    expect(wrapper).toMatchSnapshot() 
});

test('should render ExpensesSummary with multiple expenses correctly',()=>{
    const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={600} />);
    expect(wrapper).toMatchSnapshot() 
});