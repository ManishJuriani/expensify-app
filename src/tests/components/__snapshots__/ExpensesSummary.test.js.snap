// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary with multiple expenses correctly 1`] = `
<div>
  <h3>
    Viewing 
    2
     
    expenses
     totalling 
    $600.00
  </h3>
</div>
`;

exports[`should render ExpensesSummary with one expense correctly 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $500.00
  </h3>
</div>
`;
